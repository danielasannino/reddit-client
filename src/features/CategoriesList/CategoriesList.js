import React from 'react';
import { useSelector } from 'react-redux';
import { selectCategories } from '../reddit/redditSlice';
import { List, ListItem, ListItemText } from '@material-ui/core'
import { useDispatch } from 'react-redux';
import { currentTopicUpdated } from '../reddit/redditSlice';

const CategoriesListItem = ({ category }) => {
    const dispatch = useDispatch();

    const onCategoryChange = () => {
        console.log('click')
        dispatch(currentTopicUpdated(category))
    }

    return <ListItem button onClick={onCategoryChange}>
        <ListItemText primary={category} />
    </ListItem>
}

export const CategoriesList = () => {
    const categories = useSelector(selectCategories);
    const categoriesList = categories.map(category => {
        return <CategoriesListItem category={category} />
    })
    return (
        <List>
            {categoriesList}
        </List>
    )
};