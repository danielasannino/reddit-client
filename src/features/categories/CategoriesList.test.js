import { CategoriesList } from './CategoriesList';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
const { shallow } = require('enzyme');

const mockStore = configureStore([])
const store = mockStore()

describe('CategoriesList component', () => {

    it('should render with the given state from redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <CategoriesList />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})