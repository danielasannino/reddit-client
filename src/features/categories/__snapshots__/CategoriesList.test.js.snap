// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList component should render with the given state from redux store 1`] = `ShallowWrapper {}`;
